xbmc/Application.cpp

```diff
@@ -2692,9 +2692,12 @@ bool CApplication::OnMessage(CGUIMessage& message)
                                                          m_itemCurrentFile, param);
 
       // we don't want a busy dialog when switching channels
-      if (!m_itemCurrentFile->IsLiveTV())
+      if (!m_itemCurrentFile->IsLiveTV() ||
+         (!m_appPlayer.IsPlayingVideo() && !m_appPlayer.IsPlayingAudio()))
```

Comment:
Wouldn't it be better to add `m_itemCurrentFile->IsLiveTV()` for the second set of conditions otherwise it will apply to all type of content and not just pvr? Not saying it'll bring any issue but it's always better to restrict it as much as possible to avoid side effects.

Reply:
It is guaranteed that the second term (`!m_appPlayer...`) only gets evaluated if first term is false. Means second term will only be checked if  `m_itemCurrentFile->IsLiveTV()` is true.